File: ios/Classes/HelpScoutImplementation.swift
//
//  HelpScoutImplementation.swift
//  help_scout
//
//  Created by Baba Yaga on 12/20/20.
//

import Foundation
import Beacon;

 public class HelpScoutImplementation {
    init() {
    }
    var settings: HSBeaconSettings = HSBeaconSettings();

    func setupBeacon(arguments: Any) -> Void {
        let channelArgs = arguments as? Array<String>
        identifyUser(email: channelArgs?[1] ?? "", name: channelArgs?[2] ?? "");
    }
    
    func openBeacon(arguments: Any)  -> Void {
        let channelArgs = arguments as? Array<String>
        HSBeacon.open(HSBeaconSettings(beaconId: channelArgs?[0] ?? ""));
    }
    func identifyUser(email:String, name:String) -> Void {
        let user=HSBeaconUser();
        user.email = email;
        user.name = name;
        HSBeacon.login(user);
    }
}

File: android/src/main/kotlin/com/help_scout/HelpScoutImplementation.kt
package com.help_scout

import android.content.Context
import com.helpscout.beacon.Beacon
import com.helpscout.beacon.ui.BeaconActivity

class HelpScoutImplementation {

    fun setupBeacon(arguments: Any) {
        val channelArgs = arguments as List<String>;
        val beaconId: String = channelArgs[0];
        try {
            Beacon.Builder()
                    .withBeaconId(beaconId)
                    .build();
            identifyUser(email = channelArgs[1], name = channelArgs[2])
        } catch (e: UnknownError) {
            throw(e);
        }
    }

    fun openBeacon(context: Context) {
        try {
            BeaconActivity.open(context);
        } catch (e: UnknownError) {
            throw(e);
        }
    }

    private fun identifyUser(email: String, name: String) {
        try {
            Beacon.identify(email, name);
        } catch (e: UnknownError) {
            throw(e);
        }
    }
}